src/MemcachedTags/MemcachedTags.php

```diff
@@ -318,6 +321,9 @@ public function deleteKeysByTags(array $tags, $compilation = MemcachedTags::COMP
             $compilation = self::COMPILATION_OR;
         }
         $keys = $this->getKeysByTags($tags, $compilation);
+        if (is_null($keys)){
```

Comment:
Why `$keys` can be null? Could show me please an example how to get a null here?

Reply:
The tag itself is not deleted when keys of a tag is deleted. Then the tag remains without any key associated with it. It is possible then to try to call that function for a tag without any associated key. Does it make sense?

Reply:
I checked, and I see, that Method `getKeysByTags` always returns an array, and null is impossible. This is the reason why I have asked about some example of code to get a null. Could you give me please some script with the example?

Reply:
Hi @cheprasov ,
I know this might sound stupid but there is a rare case.

```     $MemcachedTags->deleteKeysByTags([]); ```

I did not like the Fatal Error in this case.

``` 
[17-Jan-2018 18:05:20 Europe/Berlin] PHP Fatal error:  Uncaught TypeError: Argument 1 passed to MemcachedTags\MemcachedTags::getTagsByKeys() must be of the type array, null given, called in /var/www/htdocs/php-memcached-tags/src/MemcachedTags/MemcachedTags.php on line 196 and defined in /var/www/htdocs/php-memcached-tags/src/MemcachedTags/MemcachedTags.php:376
Stack trace:
#0 /var/www/htdocs/php-memcached-tags/src/MemcachedTags/MemcachedTags.php(196): MemcachedTags\MemcachedTags->getTagsByKeys(NULL)
#1 /var/www/htdocs/php-memcached-tags/src/MemcachedTags/MemcachedTags.php(327): MemcachedTags\MemcachedTags->_removeKeys(NULL)
#2 /var/www/htdocs/php-memcached-tags/example.php(109): MemcachedTags\MemcachedTags->deleteKeysByTags(Array)
#3 {main}
  thrown in /var/www/htdocs/php-memcached-tags/src/MemcachedTags/MemcachedTags.php on line 376

Reply:
@nmoeini Thanks a lot, I will check.

Reply:
@nmoeini Thanks a lot, it really helped to find the error. 
Unfortunately, I will close your PR, because it does not fix the reason of the bug.
I fixed the bag in the next PR: https://github.com/cheprasov/php-memcached-tags/pull/13/files
Thanks for help.